src/modules/disk_used.rs

```diff
@@ -0,0 +1,307 @@
+use std::{
+    error::Error,
+    fmt,
+    path::{Path, PathBuf},
+};
+
+use byte_unit::{Byte, ByteError, ByteUnit};
+use sysinfo::{Disk, DiskExt, RefreshKind, System, SystemExt};
+
+use super::{Context, Module, RootModuleConfig, Shell};
+
+use crate::configs::disk_used::DiskUsedConfig;
+use crate::formatter::StringFormatter;
+use crate::segment::Segment;
+
+fn format_byte(n_byte: f64) -> Result<String, ByteError> {
+    let byte = Byte::from_unit(n_byte, ByteUnit::B)?;
+    let mut display_bytes = byte.get_appropriate_unit(false).format(0);
+    display_bytes.retain(|c| c != ' ');
+    Ok(display_bytes)
+}
+
+fn get_disk_name(disk: &'_ Disk) -> Option<&'_ str> {
+    let full_disk_name = disk.get_name();
+    let disk_name = Path::new(full_disk_name).file_name();
+    match disk_name {
+        Some(disk_name) => disk_name.to_str(),
+        None => full_disk_name.to_str(),
+    }
+}
+
+fn format_disk_used(
+    disk: &Disk,
+    config: &DiskUsedConfig,
+    show_disk_name: bool,
+    add_separator: bool,
+    percentage_char: &'_ str,
+) -> Result<Vec<Segment>, Box<dyn Error>> {
+    let used_space = (disk.get_total_space() - disk.get_available_space()) as f64;
```

Comment:
I'd prefer if these were kept as `u64` like in `memory_usage` and converted to `f64` on demand